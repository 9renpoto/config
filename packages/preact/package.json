{
  "name": "@9renpoto/preact",
  "version": "1.0.0",
  "private": true,
  "license": "MIT",
  "devDependencies": {
    "@storybook/preact": "^6.1.8"
  }
}
