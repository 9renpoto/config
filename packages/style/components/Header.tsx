import classnames from 'classnames'
import { h, Component } from 'preact'

export type Props = {
  titles: Array<string>
  baseUrl: string
}

export type State = {
  isActive: boolean
}

const NAVBAR = 'navbar'
const ID = `navMenu`

export class Header extends Component<Props, State> {
  state: State = {
    isActive: false,
  }
  render() {
    const { titles, baseUrl } = this.props
    return (
      <header className={`${NAVBAR}`}>
        <div className={`${NAVBAR}-brand`}>
          <div
            className={classnames({
              [`${NAVBAR}-burger burger`]: true,
              'is-active': this.state.isActive,
            })}
            data-target={ID}
            onClick={_ => this.setState({ isActive: !this.state.isActive })}
          >
            <span />
            <span />
            <span />
          </div>
        </div>
        <div
          className={classnames({
            [`${NAVBAR}-menu`]: true,
            'is-active': this.state.isActive,
          })}
          id={ID}
        >
          <div className={`${NAVBAR}-end`}>
            {titles.map((title, index) => (
              <a
                className={`${NAVBAR}-item`}
                href={`${baseUrl}/${title}/`}
                key={index}
              >
                {title}
              </a>
            ))}
          </div>
        </div>
      </header>
    )
  }
}
