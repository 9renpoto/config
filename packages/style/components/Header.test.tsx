import { h } from "preact";
import { render } from "preact-render-to-string";
import { Header } from "./Header";

describe("Header", () => {
  it("render", () => {
    const tree = render(<Header baseUrl="" titles={["hello"]} />);
    expect(tree).toMatchSnapshot();
  });
});
