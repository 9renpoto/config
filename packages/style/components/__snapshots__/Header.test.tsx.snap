// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render 1`] = `"<header class=\\"navbar\\"><div class=\\"navbar-brand\\"><div class=\\"navbar-burger burger\\" data-target=\\"navMenu\\"><span></span><span></span><span></span></div></div><div class=\\"navbar-menu\\" id=\\"navMenu\\"><div class=\\"navbar-end\\"><a class=\\"navbar-item\\" href=\\"/hello/\\">hello</a></div></div></header>"`;
